### ISSUE TITLE
Different value generation strategies on Owned Entity Types

### ISSUE THREAD

**Issue by jochenjonc** [dotnet/efcore#20740]
When trying Preview 3 of EF Core 5, to see if issue #18299 that has been solved, also solves my issue #19907, I got stuck on the following error.

```
'Contact.Id' and 'Contact.Location#Location.Address#Address.LocationContactId' are both mapped to column 'Id' in 'Contact' but are configured with different value generation strategies.
```

I didn't have this error on Preview 2 and on EF Core 2.2.

### Steps to reproduce

I have the following entities and owned types:

``` C#
public class Contact : Entity
{
	protected Contact()
	{
		Location = new Location();
	}

	public string Name { get; set; }
	public Location Location { get; protected set; }
}

public class Location
{
	private Address _address = new Address();

	public Address Address
	{
		get => _address;
		set => _address = value ?? throw new ArgumentNullException(nameof(Address));
	}

	public Point Coordinates { get; set; }
}

public class Address
{
	public string Street { get; set; }
	public string Number { get; set; }
	public string ZipCode { get; set; }
	public string City { get; set; }
	public string Country { get; set; }
}
```

The EntityTypeConfiguration is as follows:

``` C#
public class ContactEntityTypeConfiguration : IEntityTypeConfiguration<Contact>
{
	public void Configure(EntityTypeBuilder<Contact> builder)
	{
		builder.Property(c => c.Id)
			.UseHiLo("ContactHiLoSequence");

		builder.Property(c => c.Name)
			.IsUnicode()
			.HasMaxLength(150);

		builder.OwnsOne(c => c.Location,  l =>
		{
			l.OwnsOne(la => la.Address, la =>
			{
				la.Property(a => a.Street).HasColumnName("Street");
				la.Property(a => a.Number).HasColumnName("Number");
				la.Property(a => a.ZipCode).HasColumnName("ZipCode");
				la.Property(a => a.City).HasColumnName("City");
				la.Property(a => a.Country).HasColumnName("Country");
			}).WithOwner();

			l.Property(lc => lc.Coordinates).HasColumnName("Coordinates");
		});
	}
}
```

When fetching the Contact entity in EF Core 5 Preview 3 I get the error mentioned above.

Here is the full stacktrace:

```
[09:20:25 ERR] HTTP GET /api/contacts responded 500 in 3864.6813 ms
System.InvalidOperationException: 'Contact.Id' and 'Contact.Location#Location.Address#Address.LocationContactId' are both mapped to column 'Id' in 'Contact' but are configured with different value generation strategies.
   at Microsoft.EntityFrameworkCore.SqlServer.Internal.SqlServerModelValidator.ValidateCompatible(IProperty property, IProperty duplicateProperty, String columnName, String tableName)
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.ValidateSharedColumnsCompatibility(IReadOnlyList`1 mappedTypes, String tableName, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.SqlServer.Internal.SqlServerModelValidator.ValidateSharedColumnsCompatibility(IReadOnlyList`1 mappedTypes, String tableName, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.ValidateSharedTableCompatibility(IModel model, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.Validate(IModel model, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.SqlServer.Internal.SqlServerModelValidator.Validate(IModel model, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.ValidatingConvention.ProcessModelFinalized(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ImmediateConventionScope.OnModelFinalized(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.OnModelFinalized(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Internal.Model.FinalizeModel()
   at Microsoft.EntityFrameworkCore.ModelBuilder.FinalizeModel()
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.GetModel(DbContext context, IConventionSetBuilder conventionSetBuilder)
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.get_Model()
   at Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServicesBuilder.<>c.<TryAddCoreServices>b__7_3(IServiceProvider p)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.Set[TEntity]()
   ...
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)
   at lambda_method(Closure , IServiceProvider , Object[] )
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerActivatorProvider.<>c__DisplayClass4_0.<CreateActivator>b__0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.<>c__DisplayClass5_0.<CreateControllerFactory>g__CreateController|0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
```

### Further technical details

EF Core version: 5 Preview 3
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: NET Core 3.1.1
Operating system: Windows 10
IDE: Visual Studio 2019 16.5.4

**Comment by ajcvickers** [comment]
@jochenjonc Thanks for reporting this. As a workaround you can explicitly configure the shadow key property in the owned types:
```C#
builder.OwnsOne(c => c.Location,  l =>
{
    l.Property<int>("ContactId").UseHiLo("ContactHiLoSequence");
    l.OwnsOne(la => la.Address, la =>
    {
        la.Property<int>("LocationContactId").UseHiLo("ContactHiLoSequence");
        ...
    }).WithOwner();
    ...
});
```

**Comment by jochenjonc** [comment]
@ajcvickers with the workaround, the query that is no being produced by EF for issue #19907 is a lot better now. Thx.

**Comment by AndriySvyryd** [comment]
Fixed in bca8fa5c79b2f75c94cbd3dc1984398a5aa13a36

**Comment by mdhthahmd** [comment]
> @jochenjonc Thanks for reporting this. As a workaround you can explicitly configure the shadow key property in the owned types:
> 
> ```cs
> builder.OwnsOne(c => c.Location,  l =>
> {
>     l.Property<int>("ContactId").UseHiLo("ContactHiLoSequence");
>     l.OwnsOne(la => la.Address, la =>
>     {
>         la.Property<int>("LocationContactId").UseHiLo("ContactHiLoSequence");
>         ...
>     }).WithOwner();
>     ...
> });
> ```

Hello, i am trying to map an owned type to db model, but the owned entity type fields are missing from the database when i apply the migration to db.

so for the following the users table only has username email phone and profileimageurl
and the beneficiary and the number fields are not there.

``` cs
public class BankAccount : ValueObject
    {
        public String Number;
        public String Beneficiary;
    
        private BankAccount() { }

        public BankAccount( string number, string beneficiary)
        {
            Number = number;
            Beneficiary = beneficiary;
        }
```
and
```cs
public class User : Entity, IAggregateRoot
    {
        private string _username;
        private string _email;
        private string _phone;
        private string _profileImageUrl;

        // owned entity type
        public BankAccount BankAccount { get; private set; }

        protected User() { }

        public User( string username, string email, string phone, string profileImageUrl, BankAccount bankAccount)
        {
            _username = username;
            _email = email;
            _phone = phone;
            _profileImageUrl = profileImageUrl;
            BankAccount = bankAccount;
        }
    }
```
entity config
```cs
class UserEntityTypeConfiguration : IEntityTypeConfiguration<User>
 {
         public void Configure(EntityTypeBuilder<User> UserConfiguration)
        {
            UserConfiguration.ToTable("users", StructureContext.DEFAULT_SCHEMA);

            UserConfiguration.HasKey(o => o.Id);

            UserConfiguration.Ignore(b => b.DomainEvents);

            UserConfiguration.Property(o => o.Id)
                .UseHiLo("userseq", StructureContext.DEFAULT_SCHEMA);


            // Owned Type Mapping
            UserConfiguration.OwnsOne(s => s.BankAccount, b => {
                b.Property<int>("UserId").UseHiLo("userseq", StructureContext.DEFAULT_SCHEMA);
                b.WithOwner();
            });
           

            UserConfiguration
                .Property<string>("_username")
                .UsePropertyAccessMode(PropertyAccessMode.Field)
                .HasColumnName("Username")
                .IsRequired(true);


            UserConfiguration
                .Property<string>("_email")
                .UsePropertyAccessMode(PropertyAccessMode.Field)
                .HasColumnName("Email")
                .IsRequired(true);

            UserConfiguration
                .Property<string>("_phone")
                .UsePropertyAccessMode(PropertyAccessMode.Field)
                .HasColumnName("Phone")
                .IsRequired(true);

            UserConfiguration
                .Property<string>("_profileImageUrl")
                .UsePropertyAccessMode(PropertyAccessMode.Field)
                .HasColumnName("ProfileImageUrl")
                .IsRequired(true);
        }
}
```

Also i get the same error of ```Different value generation strategies on Owned Entity Type``` when using ``` UserConfiguration.OwnsOne(s => s.BankAccount)```.

So what could be causing ef core not to map ```BankAccount``` fields to db with the stated work around?,  how do i tell ef to map the fields in owned entity?

Thanks in advance
